Source/WebGPU/WebGPU/Sampler.mm

```diff
@@ -279,27 +252,27 @@ static uint64_t floatToUint64(float f)
 
     Locker locker { samplerStateLock };
     if (!cachedSamplerStates) {
-        cachedSamplerStates = [NSMutableDictionary dictionary];
+        cachedSamplerStates = [NSMapTable strongToWeakObjectsMapTable];
         lastAccessedKeys = [NSMutableOrderedSet orderedSet];
     }
-
     id<MTLSamplerState> samplerState = [cachedSamplerStates objectForKey:m_samplerIdentifier];
     if (samplerState)
         return samplerState;
 
     id<MTLDevice> device = m_device->device();
     if (!device)
         return nil;
-    if (cachedSamplerStates.count >= device.maxArgumentBufferSamplerCount) {
```

Comment:
The if statement was incorrect before given the below assertion, it should have been a while